docs/mkdocs/preflight/preflight-validation.md

```diff
@@ -0,0 +1,100 @@
+        // +optional
+        StatusReason string `json:"statusReason,omitempty"`
+
+        // Current stage of the verification process:
+        // image (image existence verification), build(build process verification)
+        // +required
+        // +kubebuilder:validation:Required
+        // +kubebuilder:validation:Enum=Image;Build;Sign;Requeued;Done
+        VerificationStage string `json:"verificationStage"`
+
+        // LastTransitionTime is the last time the CR status transitioned from one status to another.
+        // This should be when the underlying status changed.  If that is not known, then using the time when the API field changed is acceptable.
+        // +required
+        // +kubebuilder:validation:Required
+        // +kubebuilder:validation:Type=string
+        // +kubebuilder:validation:Format=date-time
+        LastTransitionTime metav1.Time `json:"lastTransitionTime" protobuf:"bytes,4,opt,name=lastTransitionTime"`
+}
+```
+
+
+Per each module you will see the following fields:
+1. Verification status - true or false, validated or not
+2. Status reason - verbal explanation reagrding the status. Why it is not validated, etc'
+3. Verification stage - describe the validation stage being executed (Image, Build, Sign)
+4. Last transition time - the tinme of the last update to the status
+
+
+
+### Preflight validation stages per Module
+
+On every KMM Module present in the cluster, preflight will run the following validations:
+1. [Image validation](#Image-validation-stage)
+2. [Build validation](#Build-validation-stage)
+3. [Sign validation](#Sign-validation-stage)
+
+### Image validation stage
+
+Image validation is always the first stage of the preflight validation that is being executed. In case image validation
+is successful, no other validations will be run on that specific module.
+Image validation consists of 2 stages:
+1. image existences and accessability. The code tries to access the image defined for the upgraded kernel in the module, and get its manifests
+2  verify the presence of the kernel module defined in the Module in the correct path for future modprobe execution. If this validation is successfull
+   it probably means that the kernel module was compiled with the correct linux headers. The correct path is:
+   <DirName>/lib/modules/<UpgradedKernel>/  
+
+### Build validation stage
+
+Build validation is executed only in case image validation has failed, and there is a Build section in the Module that is relevent for the
+upgraded kernel. Build validation will try to run build job and validate that it finishes successfully. In case "Push" flag is defined in the Preflight CR,
+it will also try to push the resulting image into its repo. The resulting image is take from the definition of the ContainerImage field of the Module CR
+Note: if Sign section is defined for the upgraded kernel, then the resulting image will not be ContainerImage field of the Module CR, but a temporary image name,
```

Comment:
How does `then the resulting image will not be ContainerImage field of the Module CR` goes with...

Reply:
in case of a build, when the sign section is defined, the build output will be some temporary image name, and not the final one. The final one can only be the result of a Sign success.

Reply:
Thanks for the explanation. Make sense.